cg/meta/orders/api.py

```diff
@@ -29,133 +31,35 @@
 class OrdersAPI(LimsHandler, StatusHandler):
     """Orders API for accepting new samples into the system."""
 
-    def __init__(self, lims: LimsAPI, status: Store, osticket: OsTicket = None):
+    def __init__(self, lims: LimsAPI, status: Store, osticket: OsTicket):
         super().__init__()
         self.lims = lims
         self.status = status
-        self.osticket = osticket
+        self.ticket_handler: TicketHandler = TicketHandler(osticket_api=osticket, status_db=status)
 
-    def submit(self, project: OrderType, order: dict, ticket: dict) -> dict:
+    def submit(self, project: OrderType, order_in: OrderIn, user_name: str, user_mail: str) -> dict:
         """Submit a batch of samples.
 
         Main entry point for the class towards interfaces that implements it.
         """
         try:
-            ORDER_SCHEMES[project].validate(order)
+            ORDER_SCHEMES[project].validate(order_in.dict())
         except (ValueError, TypeError) as error:
             raise OrderError(error.args[0])
-
-        self._validate_customer_on_imported_samples(project, order)
+        self._validate_customer_on_imported_samples(project=project, order=order_in)
 
         # detect manual ticket assignment
-        ticket_match = re.fullmatch(r"#([0-9]{6})", order["name"])
-
-        if ticket_match:
-            ticket_number = int(ticket_match.group(1))
-            LOG.info(f"{ticket_number}: detected ticket in order name")
-            order["ticket"] = ticket_number
-        else:
-            # open and assign ticket to order
-            try:
-                if self.osticket:
-                    message = self._create_new_ticket_message(
-                        order=order, ticket=ticket, project=project
-                    )
+        ticket_number: Optional[int] = TicketHandler.parse_ticket_number(order_in.name)
+        if not ticket_number and self.ticket_handler:
```

Comment:
Is there ever a time where self.ticket_handler is None?

Reply:
No, it was like that before this PR but after discussions with Patrik it will not happen again. Good point

Reply:
removed